src/value/decode.rs

```diff
@@ -251,27 +251,31 @@ use std::time::Duration;
 impl<'a> Into<Duration> for Value<'a> {
     fn into(self) -> Duration {
         if let ValueInner::Time(mut v) = self.0 {
-            assert!(v.len() == 8 || v.len() == 12);
+            assert!(v.len() == 0 || v.len() == 8 || v.len() == 12);
 
-            let neg = v.read_u8().unwrap();
-            if neg != 0u8 {
-                unimplemented!();
-            }
-
-            let days = u64::from(v.read_u32::<LittleEndian>().unwrap());
-            let hours = u64::from(v.read_u8().unwrap());
-            let minutes = u64::from(v.read_u8().unwrap());
-            let seconds = u64::from(v.read_u8().unwrap());
-            let micros = if v.len() == 12 {
-                v.read_u32::<LittleEndian>().unwrap()
+            if v.len() == 0 {
+                Duration::from_secs(0)
```

Comment:
Let's make this `return Duration::from_secs(0);` so that we can avoid placing the rest of the function body in an `else` block.